beacon/pow/src/main/java/tech/pegasys/teku/beacon/pow/DepositSnapshotFileLoader.java

```diff
@@ -68,11 +70,19 @@ public LoadDepositSnapshotResult loadDepositSnapshot() {
       try {
         STATUS_LOG.loadingDepositSnapshotResource(sanitizedUrl);
         final DepositTreeSnapshot depositTreeSnapshot = loadFromUrl(depositSnapshotResourceUrl);
+        // Validate
+        DepositTree.fromSnapshot(depositTreeSnapshot);
         STATUS_LOG.onDepositSnapshot(
             depositTreeSnapshot.getDepositCount(), depositTreeSnapshot.getExecutionBlockHash());
         return LoadDepositSnapshotResult.create(Optional.of(depositTreeSnapshot));
       } catch (final Exception e) {
-        LOG.warn("Failed to load deposit tree snapshot from " + sanitizedUrl, e);
+        if (ExceptionUtil.hasCause(e, IllegalArgumentException.class)) {
```

Comment:
Can we simply catch `IllegalArgumentException` and then `Exception` here, instead of using `hasCause`?

Reply:
not if it's wrapped...

Reply:
I don't think it is

Reply:
I think it's just a safer approach. Write it safe rather than guess. And something could be changed there in the future.

Reply:
fair in this case, not sure if this approach holds in general (referring to exception handling, not to "write safe" statement :) )

Reply:
ok, removing, it's redundant